### ISSUE TITLE
Support real equalMatch in OAL core

### ISSUE THREAD

**Issue by wu-sheng** [apache/skywalking#7521]
### Discussed in https://github.com/apache/skywalking/discussions/7517

<div type='discussions-op-text'>

<sup>Originally posted by **807801002** August 20, 2021</sup>
Which field can be used at filter()？I thought all fields in org.apache.skywalking.oap.server.core.source.ServiceRelation can be used to the filter expression for "from(ServiceRelation.*)".But the "componentId == 7" is error. which is a field at ServiceRelation.
![Snipaste_2021-08-20_17-20-14](https://user-images.githubusercontent.com/8719537/130212915-97919d85-aa22-4a03-a2ea-3987d55c636a.png)
</div>

We have `EqualMatch`(function name, `stringMatch`), but this is for String type only. We should add a real `equalMatch` function in grammar tree and implementation in `org.apache.skywalking.oap.server.core.analysis.metrics.expression`. 
Also, we should rename class `EqualMatch` to `StringMatch`.

**Comment by YczYanchengzhe** [comment]
I want to implement this by adding intmatch and wonder if it assign assigned to me

**Comment by kezhenxu94** [comment]
@YczYanchengzhe assigned, thanks

**Comment by YczYanchengzhe** [comment]
I encountered a problem.
After I completed the development and passed the unit test, I wanted to run the integration and found this error. The cause of the error was that the oal.core included parameter configuration could not be recognized. What could be the reason for this?


line 23:46 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 24:42 mismatched input '100' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 30:54 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 40:120 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 41:120 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 50:137 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 51:137 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 62:56 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 68:118 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
line 73:69 extraneous input '10' expecting {')', BOOL_LITERAL, NUMBER_LITERAL, IDENTIFIER}
